chips/nrf52/src/spi.rs

```diff
@@ -0,0 +1,501 @@
+use chip;
+use core::cell::Cell;
+use core::cmp;
+use core::ptr;
+use kernel::ReturnCode;
+use kernel::common::take_cell::TakeCell;
+use kernel::hil;
+use nrf5x::nvic;
+use nrf5x::peripheral_interrupts::NvicIdx;
+use nrf5x::pinmux::Pinmux;
+
+pub static mut SPIM0: SPIM = SPIM::new(0);
+pub static mut SPIM1: SPIM = SPIM::new(1);
+pub static mut SPIM2: SPIM = SPIM::new(2);
+
+mod registers {
```

Comment:
We're also (trying to) more aggressively include `///` comments towards the generation of nice documentation (see https://rustdoc--tockosorg.netlify.com/rustdoc/sam4l/adc/struct.AdcChannel.html for the working branch)

I think some simple 'what is this' comments attached to all the structures would be great - also most of the `//`'s below could probably upgrade to `///` pretty happily. The SAM4L ADC implementation linked about is a decent template for the balance we're going for (though definitely open to feedback/pushback on documentation)